### PR TITLE
Ashraf project week-2

### PR DESCRIPTION
**Pull request by Ashraf-Alshashaa**

### CODE REVIEW

File: week1/project/ecommerce/src/App.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import "./App.css";
+import { useState } from "react";
+import NavBar from "./Nav-bar";
+import Products from "./Products";
+
+// link to show the app https://grand-bienenstitch-b51d26.netlify.app
```
Comment (nick-barth) [inline week1/project/ecommerce/src/App.js:6]: still love it, this would usually go int he readme tho.

File: week2/project/ecommerce/src/App.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import "./App.css";
+import { useState } from "react";
+import NavBar from "./Nav-bar";
+import Products from "./Products";
+
+// link to show the app for week-2: https://dashing-bublanina-fbf9a3.netlify.app
```
Comment (nick-barth) [inline week2/project/ecommerce/src/App.js:6]: please only submit one HW per PR.

File: week2/project/ecommerce/src/Nav-bar.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { React, useEffect, useState } from "react";
+import { checkReturn } from "./index.js";
+
+const NavBar = (cb) => {
+  const [{ categoriesArray, status, isLoading, errorMassage }, setData] =
+    useState({
```
Comment (nick-barth) [inline week2/project/ecommerce/src/Nav-bar.js:6]: not good, not good, every time one of these key value pairs change, the entire page will be rerendered.

File: week2/project/ecommerce/src/Nav-bar.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      categoriesArray: [],
+      status: "",
+      isLoading: true,
+      errorMassage: "something wrong happened",
+    });
+
+  useEffect(() => {
+    (async () => {
```
Comment (nick-barth) [inline week2/project/ecommerce/src/Nav-bar.js:14]: cute, but not very readable, please declare a named function, and call it.

File: week2/project/ecommerce/src/Nav-bar.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  const items = categoriesArray.map((category) => {
+    return (
+      <button key={category} onClick={() => cb(category)}>
+        {category}
+      </button>
+    );
+  });
+
+  return checkReturn(isLoading, status, items, errorMassage);
```
Comment (nick-barth) [inline week2/project/ecommerce/src/Nav-bar.js:44]: You should really be returning JSX, this is difficult to read, as people don't know where the JSX in your component will be.

File: week2/project/ecommerce/src/ProductDetails.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  const items = (
+    <article className="details-container">
+      <h2>{title}</h2>
+      <section>
+        <p>{description}</p>
+        <img src={image} alt={title} />
+      </section>
+    </article>
+  );
+  return checkReturn(isLoading, status, items, errorMassage);
```
Comment (nick-barth) [inline week2/project/ecommerce/src/ProductDetails.js:48]: same issue as the other component, looks like you copy pasted this, which is okay! but when you fix one, fix both.

File: week2/project/ecommerce/src/Products.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  const items = products
+    .filter(({ category }) => category === categoryName)
+    .map(({ title, image, id }) => (
+      <div key={`container_${id}`} className="card">
+        <img src={image} alt={title} />
+        <Link className="link" to={`/products/${id}`}>
+          {title}
+        </Link>
+      </div>
+    ));
```
Comment (nick-barth) [inline week2/project/ecommerce/src/Products.js:43]: same issue again

**Review by nick-barth — CHANGES_REQUESTED** [review]

File: week1/project/ecommerce/src/App.js (hunk 1 above)
Comment (Ashraf-Alshashaa) [reply week1/project/ecommerce/src/App.js:6]: thanks, I'm happy to hear that.
I forgot to move the link to the readme file.

File: week2/project/ecommerce/src/App.js (hunk 2 above)
Comment (Ashraf-Alshashaa) [reply week2/project/ecommerce/src/App.js:5]: Sorry it happened by mistake

File: week2/project/ecommerce/src/Nav-bar.js (hunk 3 above)
Comment (Ashraf-Alshashaa) [reply week2/project/ecommerce/src/Nav-bar.js]: it's done

File: week2/project/ecommerce/src/Nav-bar.js (hunk 4 above)
Comment (Ashraf-Alshashaa) [reply week2/project/ecommerce/src/Nav-bar.js]: it's done

File: week2/project/ecommerce/src/Nav-bar.js (hunk 5 above)
Comment (Ashraf-Alshashaa) [reply week2/project/ecommerce/src/Nav-bar.js]: it's done

File: week2/project/ecommerce/src/ProductDetails.js (hunk 6 above)
Comment (Ashraf-Alshashaa) [reply week2/project/ecommerce/src/ProductDetails.js]: it's done

File: week2/project/ecommerce/src/Products.js (hunk 7 above)
Comment (Ashraf-Alshashaa) [reply week2/project/ecommerce/src/Products.js]: it's done

File: week2/project/ecommerce/src/Nav-bar.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      }
+      setData((stateObj) => {
+        return { ...stateObj, status: 404, isLoading: false };
+      });
+    })();
+  }, []);
+
+  const items = categoriesArray.map((category) => {
+    return (
+      <button key={category} onClick={() => cb(category)}>
```
[the remark this answers is not in the document]
Comment (Ashraf-Alshashaa) [reply week2/project/ecommerce/src/Nav-bar.js]: it's done

**Pull request closed without merging**